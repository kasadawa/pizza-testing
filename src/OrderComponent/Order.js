import React from 'react';


class Order extends React.Component{
    constructor(props){
        super(props);

        const ordernum = +window.sessionStorage.getItem('ordernum');
        console.log(ordernum)
         const orders = [...Array(ordernum)].map((v,key)=>{
            console.log(key)
            var tmp = window.sessionStorage.getItem('order'+(key+1));
            return JSON.parse(tmp) ; 
        });
        this.state = {orders, ordernum};
        console.log(orders)
    }

    render(){
        return (
            <div className='row justify-content-md-center'>
             <div className='col-md-8'>
                <h1 className='text-center'>Your Selection</h1>
                <br></br>
                <br></br>
                {this.state.orders.map((order,key)=>{
                    return <div key={key}>
                        <p>{order.quantity} x {order.name}  with {order.descr} </p>
                        <p> Style: {order.style || ''} </p>
                        <p> { order.options.length !== 0 ?"Options: " + order.options.join(', ')  : ''}</p>
                        <hr></hr>
                    </div>;
                })}
                </div>
            </div> );

    }
}
export default Order ;