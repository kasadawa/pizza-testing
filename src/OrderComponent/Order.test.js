import React from 'react';
import Order from './Order';
import renderer from 'react-test-renderer';

test('Order changes the class when hovered', () => {
  const component = renderer.create(
    <Order></Order>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});