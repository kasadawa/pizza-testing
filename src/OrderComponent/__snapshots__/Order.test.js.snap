// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order changes the class when hovered 1`] = `
<div
  className="row justify-content-md-center"
>
  <div
    className="col-md-8"
  >
    <h1
      className="text-center"
    >
      Your Selection
    </h1>
    <br />
    <br />
  </div>
</div>
`;
